platform/disk/ephemeral_device_partitioner.go

```diff
@@ -0,0 +1,192 @@
+package disk
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"code.cloudfoundry.org/clock"
+	bosherr "github.com/cloudfoundry/bosh-utils/errors"
+	boshlog "github.com/cloudfoundry/bosh-utils/logger"
+	boshretry "github.com/cloudfoundry/bosh-utils/retrystrategy"
+	boshsys "github.com/cloudfoundry/bosh-utils/system"
+)
+
+type EphemeralDevicePartitioner struct {
+	partedPartitioner Partitioner
+	deviceUtil        Util
+	logger            boshlog.Logger
+
+	logTag      string
+	cmdRunner   boshsys.CmdRunner
+	fs          boshsys.FileSystem
+	timeService clock.Clock
+}
+
+type Settings struct {
+	AgentID string `json:"agent_id"`
+}
+
+func NewEphemeralDevicePartitioner(
+	partedPartitioner Partitioner,
+	deviceUtil Util,
+	logger boshlog.Logger,
+	cmdRunner boshsys.CmdRunner,
+	fs boshsys.FileSystem,
+	timeService clock.Clock,
+) *EphemeralDevicePartitioner {
+	return &EphemeralDevicePartitioner{
+		partedPartitioner: partedPartitioner,
+		deviceUtil:        deviceUtil,
+		logger:            logger,
+		logTag:            "EphemeralDevicePartitioner",
+		cmdRunner:         cmdRunner,
+		fs:                fs,
+		timeService:       timeService,
+	}
+}
+
+func (p *EphemeralDevicePartitioner) Partition(devicePath string, partitions []Partition) error {
+	existingPartitions, deviceFullSizeInBytes, err := p.partedPartitioner.GetPartitions(devicePath)
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Getting existing partitions of `%s'", devicePath)
+	}
+
+	if p.matchPartitionNames(existingPartitions, partitions, deviceFullSizeInBytes) {
+		p.logger.Info(p.logTag, "%s already partitioned as expected, skipping", devicePath)
+		return nil
+	}
+
+	err = p.removePartitions(existingPartitions, devicePath)
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Removing existing partitions of `%s'", devicePath)
+	}
+
+	err = p.ensureGPTPartition(devicePath)
```

Comment:
If we understand this right, `wipefs` will completely remove all partition tables. You should therefore not need to `ensureGPT` labels later, because the `parted` partitioner will always re-create the GPT partition table if it doesn't exist.

Can you try that?